File: python/src/models/ms_my_model.py
from sqlalchemy.dialects.mysql import BIGINT, VARCHAR, TEXT, DATETIME, DOUBLE, DECIMAL

class SQLSERVER2MYSQL:
    # MySQL 데이터 타입 정의
    INT = BIGINT  # MySQL에서 INT와 BIGINT는 따로 있지만, BIGINT를 사용해 통일
    BIGINT = BIGINT
    VARCHAR = VARCHAR(1200)
    NVARCHAR = VARCHAR
    TEXT = TEXT
    DATETIME = DATETIME
    DATETIME2 = DATETIME  # MySQL에서는 DATETIME과 DATETIME2의 차이가 없으므로 같은 것으로 처리
    FLOAT = DOUBLE  # MySQL에서는 FLOAT와 DOUBLE을 분리하지만 DOUBLE로 매핑
    DECIMAL = DECIMAL

    def ms_mysql_data_mapping(self):
        mapping_data = {
            'int': SQLSERVER2MYSQL.INT,
            'bigint': SQLSERVER2MYSQL.BIGINT,
            'varchar': SQLSERVER2MYSQL.VARCHAR,
            'nvarchar': SQLSERVER2MYSQL.NVARCHAR,
            'text': SQLSERVER2MYSQL.TEXT,
            'datetime': SQLSERVER2MYSQL.DATETIME,
            'datetime2': SQLSERVER2MYSQL.DATETIME2,
            'float': SQLSERVER2MYSQL.FLOAT,
            'decimal': SQLSERVER2MYSQL.DECIMAL
        }
        return mapping_data

File: rust/src/models/ms_my_model.rs
use sqlx::types::{BigDecimal, chrono::NaiveDateTime};
use std::collections::HashMap;

#[derive(Debug)]
enum MySQLDataType {
    BigInt,
    VarChar(usize),
    Text,
    DateTime,
    Double,
    Decimal,
}

struct SQLServerToMySQL;

impl SQLServerToMySQL {
    fn ms_mysql_data_mapping() -> HashMap<&'static str, MySQLDataType> {
        let mut mapping_data = HashMap::new();

        mapping_data.insert("int", MySQLDataType::BigInt);
        mapping_data.insert("bigint", MySQLDataType::BigInt);
        mapping_data.insert("varchar", MySQLDataType::VarChar(1200));
        mapping_data.insert("nvarchar", MySQLDataType::VarChar(1200));
        mapping_data.insert("text", MySQLDataType::Text);
        mapping_data.insert("datetime", MySQLDataType::DateTime);
        mapping_data.insert("datetime2", MySQLDataType::DateTime);
        mapping_data.insert("float", MySQLDataType::Double);
        mapping_data.insert("decimal", MySQLDataType::Decimal);

        mapping_data
    }
}